Improve rendering of the tabs

jacr13 opened the pull request:
The rendering is similar in all the tabs. 

In addition, the tab disposition is the same across all the tabs.

If the translation is not present, the default names will be displayed (in English), some effort needs to be made here to add translation for all supported languages. It shouldn't be very hard to translate since they are very basic words.

This probably fixes #298. I intentionally removed the shopping tab because it requires some effort to render correctly (#136).

The header should be extended to add an extra row with different parameters to choice from (#322 and possibly #192)

benbusby commented on app/utils/search.py:
```diff
@@ -122,6 +122,7 @@ def generate_response(self) -> str:
                                self.request_params,
                                self.config,
                                content_filter.near)
+        self.full_query = full_query
```
This doesn't appear to get used anywhere

```suggestion
```

jacr13 replied:
It is used in the routes.py file when I call get_tab_contents [here](https://github.com/jacr13/whoogle-search/blob/7a86519b84bf3e232ae1268893e87e962e2b987a/app/routes.py#L263).

benbusby commented on app/utils/results.py:
```diff
@@ -222,3 +223,41 @@ def add_ip_card(html_soup: BeautifulSoup, ip: str) -> BeautifulSoup:
+
+        # update href with query
+        query = full_query
+        if '&tbm=' in full_query:
+            query = full_query.replace(f"&tbm={search_type}", '')
```
Can be simplified to

```suggestion
        query = full_query.replace(f'&tbm={search_type}', '')
```

jacr13 replied:
Good point!

benbusby commented on app/templates/header.html:
```diff
@@ -67,6 +67,25 @@
         </div>
     </header>
+    <div>
+      <div class="desktop-tap-div">
```
I assume this is a typo?

```suggestion
      <div class="desktop-tab-div">
```

benbusby commented on app/static/css/header.css:
```diff
@@ -106,3 +106,95 @@ header {
     flex: 1
 }
+
+.desktop-tap-div {
```
Typo here (mentioned below as well)

```suggestion
.desktop-tab-div {
```

benbusby commented on app/filter.py:
```diff
@@ -277,6 +288,28 @@ def update_styling(self, soup) -> None:
         except AttributeError:
             pass
 
+        # Fix body max with on images tab
```
Can you elaborate what's going on here? I'm not sure I follow.

jacr13 replied:
The max-width returned to the images tab is 700 (for body tag) which is different on all other tabs (736) so we need to replace it. The html styling is also missing and is needed to have the exact same formatting.

benbusby commented on app/filter.py:
```diff
@@ -277,6 +288,28 @@ def update_styling(self, soup) -> None:
+        div = soup.find('div', attrs={'class': "n692Zd"})
+        if style and div and not self.mobile:
+            css = style.string
+            css_html_tag = (
+                'html{'
+                'font-family: Roboto, Helvetica Neue, Arial, sans-serif;'
+                'font-size: 14px;'
+                'line-height: 20px;'
+                'text-size-adjust: 100%;'
+                'color: #3c4043;'
+                'word-wrap: break-word;'
+                'background-color: #fff;'
+                '}'
```
Why is this necessary? It doesn't seem to affect the page from what I can tell, and things like `background-color` and `color` are being ignored altogether. I tried removing it and couldn't tell any difference.

jacr13 replied:
The colors and background colors can be removed, the others are needed to have a uniform styling across the tabs (It is mainly needed to fix the text formatting). It is only used in the images tab when the view image option is disabled!

benbusby commented on app/templates/header.html:
```diff
@@ -67,6 +67,25 @@
+          </div>
+      </div>
+      <div class="" id="s">
+  </div>
```
Is there a `mobile-tab-div` in the works?

jacr13 replied:
I will work on this !

benbusby commented on app/static/settings/translations.json:
```diff
@@ -34,7 +34,13 @@
         "dark": "dark",
-        "system": "system"
+        "system": "system",
+        "all": "All",
+        "images": "Images",
+        "maps": "Maps",
+        "videos": "Videos",
+        "news": "News",
+        "books": "Books"
```
Equal translations are enforced in the tests, so these have to be translated to the existing languages. I'd like to automate this in the future, but haven't looked into it yet. 

That being said, I can take care of these. Whenever the feature is ready to merge, I'll go through and add the remaining translations before merging.

jacr13 replied:
Yes, I saw that the test didn't pass due to the keys not being the same on all languages, but I have just filled the English to have the "reference" keys.

benbusby commented on app/static/css/header.css:
```diff
@@ -106,3 +106,95 @@ header {
+}
+
+.bRsWnc {
+    background-color: #fff;
```
Is this being applied? It's likely getting overwritten by the Whoogle theme. 

```suggestion
    background-color: #fff;
```

jacr13 replied:
Maybe not, I have just taken all the css related to the tabs, I need to do some cleaning.

benbusby commented on app/static/css/header.css:
```diff
@@ -106,3 +106,95 @@ header {
+    width: 100%;
+    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+    overflow: hidden;
+}
```
From the class names, I'm guessing that these were extracted from the styles that Google includes with the tab bar. 

Would it be too much trouble to update the names of these classes to be more accurate? I'd like to avoid scenarios where we're using obfuscated class names provided by Google, since that ends up making the project harder to maintain if we need to tweak any of these later.

jacr13 replied:
Exact!

You are right, will fix it.

benbusby reviewed (changes requested):
Awesome @jacr13, thanks for working on this feature. On desktop it seems to work really well. I did a quick review, the main things are:

- Are you working on a mobile tab replacement as well? Currently the mobile tab bar is removed but not replaced by anything
- Some styling elements are potentially not necessary, or need to be updated.

jacr13 commented:
Thanks @benbusby for reviewing the code!

This is a first version and of course it needs some rework. 
For the mobile version I have not done any testing, I will do when addressing the other issues!

I will not have too much time the next two weeks, so I may take a couple of weeks to fix yours remarks and implement the rest.

jacr13 commented:
I have added support for the mobile version and fixed the issues you highlighted before.

The translation still needs to be done for all the languages (except English).

benbusby commented:
> I have added support for the mobile version and fixed the issues you highlighted before.
> 
> The translation still needs to be done for all the languages (except English).

Awesome! Will review this week.

benbusby commented:
Just FYI -- I'm holding off on reviewing this for a bit longer since Google is in the middle of changing how results are formatted, and I want to make sure their formatting changes don't break this feature. I'll be sure to review once it seems like the new result formatting changes they're making are finished/permanent.

DUOLabs333 commented:
Problem I've been having -- between switching tabs, for a brief second, it switches to some font (maybe Times New Roman), before going back to the default font.
